File: momentra/momentra/Shell/GroupActive/Shared/GroupActiveTheme.swift
import SwiftUI

/// Figma Group active tokens — 575:14165 family.
enum GroupActiveTheme {
    static let bg = Color(hex: "#131313")
    static let brand = Color(hex: "#FFB598")
    static let brandSoft = Color(hex: "#66FFB598")
    static let text = Color(hex: "#E5E2E1")
    static let secondary = Color(hex: "#DFC0B4")
    static let card = Color(hex: "#201F1F")
    static let border = Color(hex: "#2E2A28")
    static let accentOrange = Color(hex: "#FF7A3D")
}

/// Figma Trip Quick Add sheet chrome — #1C1A24 / #FF7A3D.
enum TripSheetTokens {
    static let bg = Color(hex: "#1C1A24")
    static let field = Color(hex: "#252332")
    static let border = Color(hex: "#323042")
    static let muted = Color(hex: "#9E9AA8")
    static let text = Color(hex: "#FFFFFF")
    static let accent = Color(hex: "#FF7A3D")
    static let accentEnd = Color(hex: "#FFB598")
}

enum GroupFinanceFormat {
    static func parseAmount(_ raw: String?) -> Decimal {
        guard let raw, !raw.isEmpty else { return 0 }
        let cleaned = raw.replacingOccurrences(of: ",", with: "")
        return Decimal(string: cleaned) ?? 0
    }

    static func utilizationPercent(expenseTotal: String?, budgetTotal: String?) -> Int {
        let budget = parseAmount(budgetTotal)
        guard budget > 0 else { return 0 }
        let expense = parseAmount(expenseTotal)
        let pct = (expense as NSDecimalNumber).doubleValue / (budget as NSDecimalNumber).doubleValue * 100
        return min(100, Int(pct.rounded()))
    }

    static func symbolFor(_ currencyCode: String) -> String {
        GroupTravelCurrencyCatalog.symbol(currencyCode)
    }

    static func formatMoney(_ raw: String?, currencyCode: String = "INR") -> String {
        let value = parseAmount(raw)
        if value <= 0 { return "—" }
        let prefix = symbolFor(currencyCode)
        let formatter = NumberFormatter()
        formatter.numberStyle = .decimal
        formatter.groupingSeparator = ","
        let nsValue = value as NSDecimalNumber
        formatter.maximumFractionDigits = nsValue.doubleValue.truncatingRemainder(dividingBy: 1) == 0 ? 0 : 2
        let body = formatter.string(from: value as NSDecimalNumber) ?? "0"
        return "\(prefix)\(body)"
    }

    static func compactMoney(_ raw: String?, currencyCode: String = "INR") -> String {
        let value = parseAmount(raw)
        let prefix = symbolFor(currencyCode)
        if value >= 100_000 {
            let k = (value as NSDecimalNumber).doubleValue / 1000
            return "\(prefix)\(Int(k.rounded()))K"
        }
        return formatMoney(raw, currencyCode: currencyCode)
    }

    /// Partitioned line e.g. "₹80,000 + $1,200 + €400" — no FX conversion.
    static func formatPartitionedAmounts(
        _ amounts: [(String, String)],
        compact: Bool = false,
        hide: Bool = false,
        separator: String = " + "
    ) -> String {
        if hide { return "••••" }
        let parts = amounts.compactMap { code, raw -> String? in
            let formatted = compact ? compactMoney(raw, currencyCode: code) : formatMoney(raw, currencyCode: code)
            return formatted == "—" ? nil : formatted
        }
        return parts.isEmpty ? "—" : parts.joined(separator: separator)
    }

    static func resolvePrimaryTotal(
        _ totals: [APIClient.GroupFinanceTotalsPayload],
        preferredCurrency: String? = nil
    ) -> APIClient.GroupFinanceTotalsPayload? {
        guard !totals.isEmpty else { return nil }
        if let pref = preferredCurrency,
           let match = totals.first(where: { $0.currencyCode.caseInsensitiveCompare(pref) == .orderedSame }) {
            return match
        }
        return totals.first(where: { parseAmount($0.budgetTotal) > 0 }) ?? totals.first
    }

    static func budgetPartitionLine(
        _ totals: [APIClient.GroupFinanceTotalsPayload],
        compact: Bool = false,
        hide: Bool = false
    ) -> String {
        formatPartitionedAmounts(
            totals.map { ($0.currencyCode, $0.budgetTotal ?? "0") },
            compact: compact,
            hide: hide
        )
    }

    static func expensePartitionLine(
        _ totals: [APIClient.GroupFinanceTotalsPayload],
        compact: Bool = false,
        hide: Bool = false
    ) -> String {
        formatPartitionedAmounts(
            totals.map { ($0.currencyCode, $0.expenseTotal ?? "0") },
            compact: compact,
            hide: hide
        )
    }

    static func positionsForParticipant(
        _ positions: [APIClient.GroupFinancePositionPayload],
        participantId: String?
    ) -> [APIClient.GroupFinancePositionPayload] {
        guard let participantId else { return [] }
        return positions.filter { $0.participantId == participantId }
    }

    static func viewerBalanceHeadline(
        viewer: APIClient.GroupFinancePositionPayload?,
        allPositions: [APIClient.GroupFinancePositionPayload],
        hide: Bool
    ) -> (headline: String, incl: String?) {
        guard let viewer else { return ("No balance yet", nil) }
        let primaryCode = viewer.currencyCode
        let primaryNet = parseAmount(viewer.netPosition)
        let headline: String = {
            let v = (primaryNet as NSDecimalNumber).doubleValue
            if v < 0 {
                let owed = formatMoney(String(format: "%.2f", abs(v)), currencyCode: primaryCode)
                return "You owe \(hide ? "••••" : owed)"
            }
            if v > 0 {
                let owed = formatMoney(viewer.netPosition, currencyCode: primaryCode)
                return "You are owed \(hide ? "••••" : owed)"
            }
            return "You're settled up"
        }()
        let others = positionsForParticipant(allPositions, participantId: viewer.participantId)
            .filter { $0.currencyCode.caseInsensitiveCompare(primaryCode) != .orderedSame }
            .compactMap { pos -> String? in
                let net = parseAmount(pos.netPosition)
                guard net != 0 else { return nil }
                let v = (net as NSDecimalNumber).doubleValue
                return formatMoney(String(format: "%.2f", abs(v)), currencyCode: pos.currencyCode)
            }
        let incl: String? = (others.isEmpty || hide) ? nil : "incl. \(others.joined(separator: " + "))"
        return (headline, incl)
    }

    static func groupPositionsByParticipant(
        _ positions: [APIClient.GroupFinancePositionPayload]
    ) -> [(String, [APIClient.GroupFinancePositionPayload])] {
        Dictionary(grouping: positions, by: \.participantId).map { ($0.key, $0.value) }
    }
}

/// Trip pulse destination labels from pulse widget or setup prefill.
enum TripPulseDestinations {
    static func fromWidget(_ widget: [String: AnyDecodable]?) -> [String] {
        guard let widget else { return [] }
        if let places = widget["places"]?.value as? [[String: Any]] {
            let labels = places.compactMap { $0["label"] as? String }.filter { !$0.isEmpty }
            if !labels.isEmpty { return labels }
        }
        for key in ["destinationText", "destination_text", "destination"] {
            if let raw = widget[key]?.value as? String, !raw.isEmpty { return [raw] }
        }
        return []
    }

    static func fromPrefill(_ prefill: GroupSetupPrefill?) -> [String] {
        guard let prefill else { return [] }
        let fromPlaces = (prefill.places ?? []).compactMap(\.label).filter { !$0.isEmpty }
        if !fromPlaces.isEmpty { return fromPlaces }
        if let legacy = prefill.destinationText, !legacy.isEmpty { return [legacy] }
        return []
    }

    static func heroSubtitle(places: [String], peopleCount: Int) -> String {
        if places.count > 1 { return "\(places.count) destinations · \(peopleCount) people" }
        if places.count == 1 { return "\(places.first!) · \(peopleCount) people" }
        return "Trip · \(peopleCount) people"
    }
}

struct GroupHeroHeader: View {
    let title: String
    let subtitle: String
    var meta: String? = nil

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            Text(title)
                .font(.plusJakarta(size: 22, weight: .heavy))
                .foregroundStyle(GroupActiveTheme.text)
            Text(subtitle)
                .font(.plusJakarta(size: 13))
                .foregroundStyle(GroupActiveTheme.secondary)
            if let meta {
                Text(meta)
                    .font(.plusJakarta(size: 12, weight: .semibold))
                    .foregroundStyle(GroupActiveTheme.brand)
            }
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(20)
        .background(
            LinearGradient(
                colors: [Color(hex: "#3D2A24"), GroupActiveTheme.bg, Color(hex: "#1A1512")],
                startPoint: .topLeading,
                endPoint: .bottomTrailing
            )
        )
        .clipShape(RoundedRectangle(cornerRadius: 24))
    }
}

struct GroupSectionCard<Content: View, Badge: View>: View {
    let title: String
    @ViewBuilder var badge: () -> Badge
    @ViewBuilder var content: () -> Content

    init(title: String, @ViewBuilder badge: @escaping () -> Badge = { EmptyView() }, @ViewBuilder content: @escaping () -> Content) {
        self.title = title
        self.badge = badge
        self.content = content
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            HStack {
                Text(title)
                    .font(.plusJakarta(size: 15, weight: .bold))
                    .foregroundStyle(GroupActiveTheme.text)
                Spacer()
                badge()
            }
            content()
        }
        .padding(16)
        .background(GroupActiveTheme.card)
        .clipShape(RoundedRectangle(cornerRadius: 18))
        .overlay(RoundedRectangle(cornerRadius: 18).stroke(GroupActiveTheme.border))
    }
}

struct GroupComingSoonBadge: View {
    var body: some View {
        Text("Coming Soon")
            .font(.plusJakarta(size: 10, weight: .bold))
            .foregroundStyle(GroupActiveTheme.brand)
            .padding(.horizontal, 10)
            .padding(.vertical, 4)
            .background(GroupActiveTheme.brandSoft)
            .clipShape(Capsule())
    }
}

struct GroupApiGapBadge: View {
    var body: some View {
        Text("API_GAP")
            .font(.plusJakarta(size: 9, weight: .bold))
            .foregroundStyle(Color(hex: "#F87171"))
            .padding(.horizontal, 8)
            .padding(.vertical, 4)
            .background(Color(hex: "#33F87171"))
            .clipShape(Capsule())
    }
}

struct GroupEmptySection: View {
    let message: String
    let detail: String

    var body: some View {
        VStack(alignment: .leading, spacing: 4) {
            Text(message)
                .font(.plusJakarta(size: 13, weight: .semibold))
                .foregroundStyle(GroupActiveTheme.text)
            Text(detail)
                .font(.plusJakarta(size: 12))
                .foregroundStyle(GroupActiveTheme.secondary)
        }
    }
}

struct GroupMetricTile: View {
    let label: String
    let value: String

    var body: some View {
        VStack(alignment: .leading, spacing: 4) {
            Text(label)
                .font(.plusJakarta(size: 11))
                .foregroundStyle(GroupActiveTheme.secondary)
            Text(value)
                .font(.plusJakarta(size: 16, weight: .bold))
                .foregroundStyle(GroupActiveTheme.text)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(12)
        .background(Color(hex: "#181716"))
        .clipShape(RoundedRectangle(cornerRadius: 14, style: .continuous))
    }
}

struct GroupQuickChip: View {
    let label: String
    let enabled: Bool
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            Text(label)
                .font(.plusJakarta(size: 12, weight: .semibold))
                .foregroundStyle(GroupActiveTheme.text)
                .padding(.horizontal, 14)
                .padding(.vertical, 8)
                .background(Color(hex: "#33FFB598"))
                .clipShape(Capsule())
                .overlay(Capsule().stroke(GroupActiveTheme.border))
        }
        .buttonStyle(.plain)
        .disabled(!enabled)
        .opacity(enabled ? 1 : 0.45)
    }
}

struct GroupProgressRing: View {
    let percent: Int
    let centerLabel: String
    let centerSub: String
    @State private var pulse = false

    var body: some View {
        ZStack {
            Circle()
                .stroke(Color(hex: "#2A2624"), lineWidth: 10)
            Circle()
                .trim(from: 0, to: CGFloat(min(max(percent, 0), 100)) / 100)
                .stroke(GroupActiveTheme.brand, style: StrokeStyle(lineWidth: 10, lineCap: .round))
                .rotationEffect(.degrees(-90))
            VStack(spacing: 2) {
                Text(centerLabel)
                    .font(.plusJakarta(size: 22, weight: .heavy))
                    .foregroundStyle(GroupActiveTheme.text)
                Text(centerSub)
                    .font(.plusJakarta(size: 11))
                    .foregroundStyle(GroupActiveTheme.secondary)
            }
        }
        .frame(width: 120, height: 120)
        .scaleEffect(pulse ? 1 : 0.94)
        .animation(.easeInOut(duration: 1.4).repeatForever(autoreverses: true), value: pulse)
        .onAppear { pulse = true }
    }
}

struct GroupProgressBar: View {
    let percent: Int

    var body: some View {
        GeometryReader { geo in
            ZStack(alignment: .leading) {
                Capsule().fill(Color(hex: "#2A2624"))
                Capsule()
                    .fill(GroupActiveTheme.brand)
                    .frame(width: geo.size.width * CGFloat(min(max(percent, 0), 100)) / 100)
            }
        }
        .frame(height: 8)
    }
}

struct GroupCtaButton: View {
    let label: String
    let enabled: Bool
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            Text(label)
                .font(.plusJakarta(size: 14, weight: .bold))
                .foregroundStyle(Color(hex: "#131313"))
                .frame(maxWidth: .infinity)
                .padding(.vertical, 14)
                .background(
                    LinearGradient(
                        colors: enabled ? [GroupActiveTheme.brand, Color(hex: "#E89574")] : [Color(hex: "#3A3533"), Color(hex: "#3A3533")],
                        startPoint: .leading,
                        endPoint: .trailing
                    )
                )
                .clipShape(RoundedRectangle(cornerRadius: 14))
        }
        .buttonStyle(.plain)
        .disabled(!enabled)
        .opacity(enabled ? 1 : 0.45)
    }
}

File: apk/app/src/main/java/com/example/momentra/ui/shell/group/shared/GroupActiveTheme.kt
package com.example.momentra.ui.shell.group.shared

import androidx.compose.animation.core.LinearEasing
import androidx.compose.animation.core.RepeatMode
import androidx.compose.animation.core.animateFloat
import androidx.compose.animation.core.infiniteRepeatable
import androidx.compose.animation.core.rememberInfiniteTransition
import androidx.compose.animation.core.tween
import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.draw.clip
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.geometry.Size
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.StrokeCap
import androidx.compose.ui.graphics.drawscope.Stroke
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.momentra.data.security.BalanceMask
import com.example.momentra.ui.theme.PlusJakartaSans
import java.math.BigDecimal
import java.math.RoundingMode
import java.text.DecimalFormat
import java.text.DecimalFormatSymbols
import java.util.Locale
import kotlin.math.min

/** Figma Group active tokens — 575:14165 family. */
object GroupActiveTheme {
    val Bg = Color(0xFF131313)
    val Brand = Color(0xFFFFB598)
    val BrandSoft = Color(0x66FFB598)
    val Text = Color(0xFFE5E2E1)
    val Secondary = Color(0xFFDFC0B4)
    val Card = Color(0xFF201F1F)
    val Border = Color(0xFF2E2A28)
    val AccentOrange = Color(0xFFFF7A3D)
    val CardRadius = 18.dp
    val HeroGradient = Brush.linearGradient(
        colors = listOf(Color(0xFF3D2A24), Color(0xFF131313), Color(0xFF1A1512)),
    )
    val ChipGradient = Brush.horizontalGradient(listOf(Color(0x33FFB598), Color(0x221A1512)))
}

/** Trip sheet / finance chrome — Figma 1257:8668 family. */
object TripSheetTokens {
    val Bg = Color(0xFF1C1A24)
    val Field = Color(0xFF252332)
    val Border = Color(0xFF323042)
    val Muted = Color(0xFF9E9AA8)
    val Text = Color(0xFFFFFFFF)
    val Accent = Color(0xFFFF7A3D) // Figma orange CTA
    val AccentEnd = Color(0xFFFFB598)
    val CardRadius = 18.dp
}

object GroupFinanceFormat {
    fun parseAmount(raw: String?): BigDecimal = try {
        raw?.replace(",", "")?.toBigDecimal() ?: BigDecimal.ZERO
    } catch (_: Exception) {
        BigDecimal.ZERO
    }

    fun utilizationPercent(expenseTotal: String?, budgetTotal: String?): Int {
        val budget = parseAmount(budgetTotal)
        if (budget <= BigDecimal.ZERO) return 0
        val expense = parseAmount(expenseTotal)
        return min(
            100,
            expense.multiply(BigDecimal(100))
                .divide(budget, 0, RoundingMode.HALF_UP)
                .toInt(),
        )
    }

    fun symbolFor(currencyCode: String): String =
        GroupTravelCurrencyCatalog.symbol(currencyCode)

    fun formatMoney(raw: String?, currencyCode: String = "INR"): String {
        val value = parseAmount(raw)
        if (value <= BigDecimal.ZERO && (raw.isNullOrBlank() || raw == "0" || raw.startsWith("0."))) {
            return "—"
        }
        val prefix = symbolFor(currencyCode)
        val symbols = DecimalFormatSymbols(Locale.US).apply { groupingSeparator = ',' }
        val pattern = if (value.stripTrailingZeros().scale() <= 0) "#,##0" else "#,##0.##"
        return prefix + DecimalFormat(pattern, symbols).format(value)
    }

    fun compactMoney(raw: String?, currencyCode: String = "INR"): String {
        val value = parseAmount(raw)
        val prefix = symbolFor(currencyCode)
        return when {
            value >= BigDecimal(100000) -> {
                val k = value.divide(BigDecimal(1000), 0, RoundingMode.HALF_UP)
                "${prefix}${k}K"
            }
            value > BigDecimal.ZERO -> formatMoney(raw, currencyCode)
            else -> "—"
        }
    }

    /** Partitioned line e.g. "₹80,000 + $1,200 + €400" — no FX conversion. */
    fun formatPartitionedAmounts(
        amounts: List<Pair<String, String>>,
        compact: Boolean = false,
        hide: Boolean = false,
        separator: String = " + ",
    ): String {
        if (hide) return "••••"
        val parts = amounts.mapNotNull { (code, raw) ->
            val formatted = if (compact) compactMoney(raw, code) else formatMoney(raw, code)
            formatted.takeIf { it != "—" }
        }
        return parts.joinToString(separator).ifBlank { "—" }
    }

    fun resolvePrimaryTotal(
        totals: List<com.example.momentra.data.api.GroupFinanceTotalDto>,
        preferredCurrency: String? = null,
    ): com.example.momentra.data.api.GroupFinanceTotalDto? {
        if (totals.isEmpty()) return null
        preferredCurrency?.let { pref ->
            totals.firstOrNull { it.currencyCode.equals(pref, ignoreCase = true) }?.let { return it }
        }
        return totals.firstOrNull { parseAmount(it.budgetTotal) > BigDecimal.ZERO } ?: totals.first()
    }

    fun budgetPartitionLine(
        totals: List<com.example.momentra.data.api.GroupFinanceTotalDto>,
        compact: Boolean = false,
        hide: Boolean = false,
    ): String = formatPartitionedAmounts(
        totals.map { it.currencyCode to (it.budgetTotal ?: "0") },
        compact = compact,
        hide = hide,
    )

    fun expensePartitionLine(
        totals: List<com.example.momentra.data.api.GroupFinanceTotalDto>,
        compact: Boolean = false,
        hide: Boolean = false,
    ): String = formatPartitionedAmounts(
        totals.map { it.currencyCode to (it.expenseTotal ?: "0") },
        compact = compact,
        hide = hide,
    )

    fun positionsForParticipant(
        positions: List<com.example.momentra.data.api.GroupFinancePositionDto>,
        participantId: String?,
    ): List<com.example.momentra.data.api.GroupFinancePositionDto> {
        if (participantId.isNullOrBlank()) return emptyList()
        return positions.filter { it.participantId == participantId }
    }

    fun viewerBalanceHeadline(
        viewer: com.example.momentra.data.api.GroupFinancePositionDto?,
        allPositions: List<com.example.momentra.data.api.GroupFinancePositionDto>,
        hide: Boolean,
    ): Pair<String, String?> {
        if (viewer == null) return "No balance yet" to null
        val primaryCode = viewer.currencyCode
        val primaryNet = parseAmount(viewer.netPosition)
        val headline = when {
            primaryNet < BigDecimal.ZERO ->
                "You owe ${BalanceMask.mask(formatMoney(primaryNet.abs().toPlainString(), primaryCode), hide)}"
            primaryNet > BigDecimal.ZERO ->
                "You are owed ${BalanceMask.mask(formatMoney(viewer.netPosition, primaryCode), hide)}"
            else -> "You're settled up"
        }
        val others = positionsForParticipant(allPositions, viewer.participantId)
            .filter { !it.currencyCode.equals(primaryCode, ignoreCase = true) }
            .mapNotNull { pos ->
                val net = parseAmount(pos.netPosition)
                if (net.compareTo(BigDecimal.ZERO) == 0) null
                else formatMoney(net.abs().toPlainString(), pos.currencyCode)
            }
        val incl = if (others.isEmpty() || hide) {
            null
        } else {
            "incl. ${others.joinToString(" + ")}"
        }
        return headline to incl
    }

    fun groupPositionsByParticipant(
        positions: List<com.example.momentra.data.api.GroupFinancePositionDto>,
    ): List<Pair<String, List<com.example.momentra.data.api.GroupFinancePositionDto>>> =
        positions.groupBy { it.participantId }.toList()
}

/** Trip pulse destination labels from pulse widget or setup prefill. */
object TripPulseDestinations {
    fun fromWidget(widget: Map<String, Any?>?): List<String> {
        if (widget == null) return emptyList()
        @Suppress("UNCHECKED_CAST")
        val placesRaw = widget["places"] as? List<*>
        val fromPlaces = placesRaw?.mapNotNull { item ->
            when (item) {
                is Map<*, *> -> item["label"]?.toString()?.takeIf { it.isNotBlank() }
                else -> null
            }
        }.orEmpty()
        if (fromPlaces.isNotEmpty()) return fromPlaces
        for (key in listOf("destinationText", "destination_text", "destination")) {
            widget[key]?.toString()?.takeIf { it.isNotBlank() && it != "null" }?.let { return listOf(it) }
        }
        return emptyList()
    }

    fun fromPrefill(prefill: com.example.momentra.data.api.GroupSetupPrefillDto?): List<String> {
        if (prefill == null) return emptyList()
        val fromPlaces = prefill.places.orEmpty().mapNotNull { it.label?.takeIf { l -> l.isNotBlank() } }
        if (fromPlaces.isNotEmpty()) return fromPlaces
        return prefill.destinationText?.takeIf { it.isNotBlank() }?.let { listOf(it) }.orEmpty()
    }

    fun heroSubtitle(places: List<String>, peopleCount: Int): String = when {
        places.size > 1 -> "${places.size} destinations · $peopleCount people"
        places.size == 1 -> "${places.first()} · $peopleCount people"
        else -> "Trip · $peopleCount people"
    }
}

@Composable
fun GroupActiveLoading(modifier: Modifier = Modifier) {
    Box(modifier.background(GroupActiveTheme.Bg), contentAlignment = Alignment.Center) {
        CircularProgressIndicator(color = GroupActiveTheme.Brand)
    }
}

@Composable
fun GroupHeroHeader(
    title: String,
    subtitle: String,
    meta: String? = null,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(24.dp))
            .background(GroupActiveTheme.HeroGradient)
            .padding(20.dp),
        verticalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        Text(
            title,
            color = GroupActiveTheme.Text,
            fontSize = 22.sp,
            fontWeight = FontWeight.ExtraBold,
            fontFamily = PlusJakartaSans,
        )
        Text(subtitle, color = GroupActiveTheme.Secondary, fontSize = 13.sp, fontFamily = PlusJakartaSans)
        meta?.let {
            Text(it, color = GroupActiveTheme.Brand, fontSize = 12.sp, fontWeight = FontWeight.SemiBold, fontFamily = PlusJakartaSans)
        }
    }
}

@Composable
fun GroupSectionCard(
    title: String,
    modifier: Modifier = Modifier,
    badge: (@Composable () -> Unit)? = null,
    content: @Composable () -> Unit,
) {
    Column(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(18.dp))
            .background(GroupActiveTheme.Card)
            .border(1.dp, GroupActiveTheme.Border, RoundedCornerShape(18.dp))
            .padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Text(
                title,
                color = GroupActiveTheme.Text,
                fontSize = 15.sp,
                fontWeight = FontWeight.Bold,
                fontFamily = PlusJakartaSans,
            )
            badge?.invoke()
        }
        content()
    }
}

@Composable
fun GroupComingSoonBadge(modifier: Modifier = Modifier) {
    Text(
        "Coming Soon",
        modifier = modifier
            .clip(RoundedCornerShape(100.dp))
            .background(GroupActiveTheme.BrandSoft)
            .padding(horizontal = 10.dp, vertical = 4.dp),
        color = GroupActiveTheme.Brand,
        fontSize = 10.sp,
        fontWeight = FontWeight.Bold,
        fontFamily = PlusJakartaSans,
    )
}

@Composable
fun GroupApiGapBadge(modifier: Modifier = Modifier) {
    Text(
        "API_GAP",
        modifier = modifier
            .clip(RoundedCornerShape(100.dp))
            .background(Color(0x33F87171))
            .padding(horizontal = 8.dp, vertical = 4.dp),
        color = Color(0xFFF87171),
        fontSize = 9.sp,
        fontWeight = FontWeight.Bold,
        fontFamily = PlusJakartaSans,
    )
}

@Composable
fun GroupEmptySection(message: String, detail: String, modifier: Modifier = Modifier) {
    Column(
        modifier = modifier.fillMaxWidth(),
        verticalArrangement = Arrangement.spacedBy(4.dp),
    ) {
        Text(message, color = GroupActiveTheme.Text, fontSize = 13.sp, fontWeight = FontWeight.SemiBold, fontFamily = PlusJakartaSans)
        Text(detail, color = GroupActiveTheme.Secondary, fontSize = 12.sp, fontFamily = PlusJakartaSans)
    }
}

@Composable
fun GroupMetricTile(
    label: String,
    value: String,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier = modifier
            .clip(RoundedCornerShape(14.dp))
            .background(Color(0xFF181716))
            .padding(12.dp),
        verticalArrangement = Arrangement.spacedBy(4.dp),
    ) {
        Text(label, color = GroupActiveTheme.Secondary, fontSize = 11.sp, fontFamily = PlusJakartaSans)
        Text(value, color = GroupActiveTheme.Text, fontSize = 16.sp, fontWeight = FontWeight.Bold, fontFamily = PlusJakartaSans)
    }
}

@Composable
fun GroupQuickChip(
    label: String,
    enabled: Boolean,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    Box(
        modifier = modifier
            .alpha(if (enabled) 1f else 0.45f)
            .clip(RoundedCornerShape(100.dp))
            .background(GroupActiveTheme.ChipGradient)
            .border(1.dp, GroupActiveTheme.Border, RoundedCornerShape(100.dp))
            .then(if (enabled) Modifier.clickable(onClick = onClick) else Modifier)
            .padding(horizontal = 14.dp, vertical = 8.dp),
    ) {
        Text(label, color = GroupActiveTheme.Text, fontSize = 12.sp, fontWeight = FontWeight.SemiBold, fontFamily = PlusJakartaSans)
    }
}

@Composable
fun GroupProgressRing(
    percent: Int,
    centerLabel: String,
    centerSub: String,
    modifier: Modifier = Modifier,
    animate: Boolean = true,
) {
    val displayPercent = percent.coerceIn(0, 100)
    val infinite = rememberInfiniteTransition(label = "pulseRing")
    val pulse by infinite.animateFloat(
        initialValue = 0.92f,
        targetValue = 1f,
        animationSpec = infiniteRepeatable(tween(1400, easing = LinearEasing), RepeatMode.Reverse),
        label = "pulseScale",
    )
    val scale = if (animate) pulse else 1f
    Box(modifier.size(120.dp), contentAlignment = Alignment.Center) {
        Canvas(modifier = Modifier.size((120 * scale).dp)) {
            val stroke = 10.dp.toPx()
            val pad = stroke / 2
            drawArc(
                color = Color(0xFF2A2624),
                startAngle = -90f,
                sweepAngle = 360f,
                useCenter = false,
                topLeft = Offset(pad, pad),
                size = Size(size.width - stroke, size.height - stroke),
                style = Stroke(width = stroke, cap = StrokeCap.Round),
            )
            if (displayPercent > 0) {
                drawArc(
                    color = GroupActiveTheme.Brand,
                    startAngle = -90f,
                    sweepAngle = 360f * (displayPercent / 100f),
                    useCenter = false,
                    topLeft = Offset(pad, pad),
                    size = Size(size.width - stroke, size.height - stroke),
                    style = Stroke(width = stroke, cap = StrokeCap.Round),
                )
            }
        }
        Column(horizontalAlignment = Alignment.CenterHorizontally) {
            Text(centerLabel, color = GroupActiveTheme.Text, fontSize = 22.sp, fontWeight = FontWeight.ExtraBold, fontFamily = PlusJakartaSans)
            Text(centerSub, color = GroupActiveTheme.Secondary, fontSize = 11.sp, fontFamily = PlusJakartaSans)
        }
    }
}

@Composable
fun GroupProgressBar(percent: Int, modifier: Modifier = Modifier) {
    val p = percent.coerceIn(0, 100)
    Box(
        modifier = modifier
            .fillMaxWidth()
            .height(8.dp)
            .clip(RoundedCornerShape(100.dp))
            .background(Color(0xFF2A2624)),
    ) {
        Box(
            modifier = Modifier
                .fillMaxWidth(p / 100f)
                .height(8.dp)
                .clip(RoundedCornerShape(100.dp))
                .background(GroupActiveTheme.Brand),
        )
    }
}

@Composable
fun GroupPulsingIcon(
    glyph: String,
    modifier: Modifier = Modifier,
) {
    val infinite = rememberInfiniteTransition(label = "settingsPulse")
    val alpha by infinite.animateFloat(
        initialValue = 0.55f,
        targetValue = 1f,
        animationSpec = infiniteRepeatable(tween(900, easing = LinearEasing), RepeatMode.Reverse),
        label = "iconAlpha",
    )
    Box(
        modifier = modifier
            .size(36.dp)
            .clip(CircleShape)
            .background(GroupActiveTheme.Card)
            .border(1.dp, GroupActiveTheme.Border, CircleShape),
        contentAlignment = Alignment.Center,
    ) {
        Text(glyph, color = GroupActiveTheme.Brand.copy(alpha = alpha), fontSize = 16.sp)
    }
}

@Composable
fun GroupCtaButton(
    label: String,
    enabled: Boolean,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    Box(
        modifier = modifier
            .fillMaxWidth()
            .alpha(if (enabled) 1f else 0.45f)
            .clip(RoundedCornerShape(14.dp))
            .background(
                if (enabled) Brush.horizontalGradient(listOf(GroupActiveTheme.Brand, Color(0xFFE89574)))
                else Brush.horizontalGradient(listOf(Color(0xFF3A3533), Color(0xFF3A3533))),
            )
            .then(if (enabled) Modifier.clickable(onClick = onClick) else Modifier)
            .padding(vertical = 14.dp),
        contentAlignment = Alignment.Center,
    ) {
        Text(label, color = Color(0xFF131313), fontWeight = FontWeight.Bold, fontSize = 14.sp, fontFamily = PlusJakartaSans)
    }
}
